examples/two-columns-layout/src/plugins/productsPlugin.tsx

```diff
@@ -0,0 +1,154 @@
+/** @jsx h */
+import {
+  AutocompletePlugin,
+  getAlgoliaResults,
+} from '@algolia/autocomplete-js';
+import { h } from 'preact';
+import { useCallback, useState } from 'preact/hooks';
+
+import { Blurhash } from '../components/Blurhash';
+import {
+  StarOutlineIcon,
+  StarFillIcon,
+  FavoriteFillIcon,
+  FavoriteOutlineIcon,
+} from '../components/Icons';
+import { ALGOLIA_PRODUCTS_INDEX_NAME } from '../constants';
+import { searchClient } from '../searchClient';
+import { ProductHit } from '../types';
+
+export interface ProductsPluginContext {
+  facetName: string;
+  facetValue: string;
+  query: string;
+}
+
+export const productsPlugin: AutocompletePlugin<ProductHit, {}> = {
+  getSources({ query }) {
+    if (!query) {
+      return [];
+    }
+
+    return [
+      {
+        sourceId: 'productsPlugin',
+        getItems() {
+          return getAlgoliaResults<ProductHit>({
+            searchClient,
+            queries: [
+              {
+                indexName: ALGOLIA_PRODUCTS_INDEX_NAME,
+                query,
+                params: {
+                  hitsPerPage: 4,
+                },
+              },
+            ],
+          });
+        },
+        onSelect({ setIsOpen }) {
+          setIsOpen(true);
+        },
+        templates: {
+          header() {
+            return <div className="aa-SourceHeaderTitle">Products</div>;
+          },
+          item({ item }) {
+            return <ProductItem hit={item} />;
+          },
+        },
+      },
+    ];
+  },
+};
+
+const formatPrice = (val: number) => {
+  return val.toFixed(2).toLocaleString().replace(/\./g, ',');
+};
```

Comment:
This mixes two notations, because `toLocaleString` uses the browser's localization to format the value, and the `replace` forces commas as decimal separators. On my French browser it works, but U.S. viewers will see some unorthodox notation.

Since the demos is in English and we usually go for American English everywhere, let's settle with U.S. notation, wdyt?

```suggestion
function formatPrice(value: number) {
  return value.toFixed(2).toLocaleString('en-US', { style: 'currency', currency: 'EUR' });
};
```

Reply:
Thanks for the explanation!